### PR TITLE
Adds RedwoodRecord docs

### PR DESCRIPTION
**Pull request by cannikin**
## WAIT FOR MERGE

Requires that https://github.com/redwoodjs/redwood/pull/3734 be merged first!

### CODE REVIEW

File: docs/redwoodRecord.md [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+└───────────┘                                  └────────────┘
+```
+
+Now we're back to one-to-many relationships. In Prisma this join table is created and maintained for you. It will be named `_CategoryToPost` and the foreign keys will simply be named `A` and `B` and point to the two separate tables. Prisma refers to this as an [implicit many-to-many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) relationship.
+
+If you want to create the join table yourself and potentially store additional data there (like a timestamp of when the product was categorized) then this is simply a one-to-many relationship on both sides: a Product has many ProductCategories and a Category has many ProductCategories. Prisma refers to this as an [explicitly many-to-many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#explicit-many-to-many-relations) relationship.
+
+> TODO: We'll be adding logic soon that will let you get to the categories from a product record (and vice versa) in explicit many-to-manys without having to manually go through ProductCategory. From this:
```
Comment (jtoar) [inline docs/redwoodRecord.md:56]: Is the "s" on "many-to-manys" a typo? Feel free to ignore if not!

**Comment by jtoar** [comment]
@cannikin Ran spell check over the file and caught a few mistakes (committed in https://github.com/redwoodjs/redwoodjs.com/pull/889/commits/f24c82e563e4a738cb2ced347ba4dfa3bfd3d5c9).

File: docs/redwoodRecord.md [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+```javascript
+const user = User.find(123)
+await user.destroy()
+// or
+await user.destroy({ throw: true })
+```
+
+### Relationships
+
+As shown in [Background and Terminology](background-and-terminology) above, RedwoodRecord provides a way to get data from one related models. For example, to get the posts belonging to a user via what we call a *relation proxy*:
```
Comment (jtoar) [inline docs/redwoodRecord.md:304]: The "one related models" bit sounds a little awkward

File: docs/cliCommands.md [hunk 3]
```diff
@@ -1472,6 +1507,28 @@ It'll also re-seed your database by automatically running the `db seed` command.
...
+```
+
+## setup
+
+Initialize project config and install packages
+
+```
+yarn redwood setup <command>
+```
+
```
Comment (jtoar) [inline docs/cliCommands.md:1531]: Did you accidentally repeat this section or is `setup` another record command?

**Review by jtoar — COMMENTED** [review]
@cannikin Looking great! Left a few comments on some things I wasn't sure about.

File: docs/redwoodRecord.md (hunk 2 above)
Comment (cannikin) [reply docs/redwoodRecord.md:304]: Yeah should just be "related models"

**Review by jtoar — APPROVED** [review]

**Pull request merged**